src/host.ps1

```diff
@@ -75,88 +132,59 @@ function Get-XoHost {
-                    $processLimit = if ($Limit -gt 0) { [Math]::Min($Limit, $hostUrls.Count) } else { $hostUrls.Count }
-                    $processUrls = $hostUrls | Select-Object -First $processLimit
-                    
-                    foreach ($hostUrl in $processUrls) {
-                        # Skip null or empty URLs
-                        if ([string]::IsNullOrEmpty($hostUrl)) {
-                            continue
-                        }
-                        
-                        try {
-                            # Extract the ID from the URL string
-                            $match = [regex]::Match($hostUrl, "\/rest\/v0\/hosts\/([^\/]+)$")
-                            if ($match.Success) {
-                                $id = $match.Groups[1].Value
-                                if (![string]::IsNullOrEmpty($id)) {
-                                    $hostDetail = Invoke-RestMethod -Uri "$script:XoHost$hostUrl" @script:XoRestParameters
 
-                                    if ($hostDetail) {
-                                        # Parse the response if it's a string
-                                        if ($hostDetail -is [string]) {
-                                            $hostDetail = $hostDetail | ConvertFrom-Json -AsHashtable
-                                        }
-                                        
-                                        ConvertTo-XoHostObject $hostDetail
-                                    }
-                                }
-                                else {
-                                    Write-Warning "Failed to extract valid ID from URL: $hostUrl"
-                                }
-                            }
-                        } catch {
-                            Write-Warning "Failed to retrieve host details from URL: $hostUrl. Error: $_"
-                        }
-                    }
-                }
-                else {
-                    Write-Verbose "No hosts found"
-                }
-            }
-            catch {
-                Write-Error "Failed to list hosts. Error: $_"
+    try {
+        $uri = "$script:XoHost/rest/v0/hosts"
+        $hostsResponse = Invoke-RestMethod -Uri $uri @script:XoRestParameters -Body ($params | ConvertTo-Json -Compress) -Method Get
+        
+        if (!$hostsResponse -or $hostsResponse.Count -eq 0) {
+            Write-Verbose "No hosts found"
+            return
+        }
+
+        $hostsToProcess = $hostsResponse
+        if ($Limit -gt 0 -and $hostsResponse.Count -gt $Limit) {
+            $hostsToProcess = $hostsResponse[0..($Limit-1)]
+        }
+
+        foreach ($hostDetail in $hostsToProcess) {
+            $fullHostDetail = Get-XoHostDetailFromPath -HostDetail $hostDetail
+            if ($fullHostDetail) {
+                $hostObj = ConvertTo-XoHostObject -InputObject $fullHostDetail
+                Write-Output $hostObj
             }
         }
```

Comment:
@iButcat Again, why query every item in a loop instead of passing the field list and limit 
directly? With this you will generate one JSON request per command + another request for every single object fetched.